Tempura/UITests/UITests.swift

```diff
@@ -0,0 +1,258 @@
+      }
+    }
+  }
+  
+  /// Hooks can be used to customize the behaviour of the mocked view controller that renders the view.
+  public enum Hook: Int {
+    /**
+     The navigation controller has been created and can be customized.
+     This hook is triggered only when the container is `navigationController`
+     */
+    case navigationControllerHasBeenCreated
+    
+    /// UIViewController's lifecycle hooks
+    case viewDidLoad
+    case viewWillAppear
+    case viewDidAppear
+    case viewDidLayoutSubviews
+  }
+  
+  /// Closure invoked when a hook is triggered
+  public typealias HookClosure<V: ViewControllerModellableView> = (V) -> Void
+  
+  /// The container UIViewController subclass in which the view can be embedded
+  public enum Container {
+    /// No container
+    case none
+    /// UINavigationController
+    case navigationController
+    /// UITabBarController
+    case tabBarController
+  }
+  
+  /// Create a snapshot image of the view and pass the test
+  /// The snapshot will be saved under the UI_TEST_DIR specified in your info.plist
+  static func verifyView(view: UIView, description: String) {
+    let recording: Bool = (Bundle.main.infoDictionary?["UI_TEST_RECORDING"] as? Bool) == true
+    snapshot(view: view, recording: recording, description: description)
+    XCTAssertTrue(true)
+  }
+  
+  /// Create a snapshot image of the view.
+  /// The snapshot will be saved under the UI_TEST_DIR specified in your info.plist
+  private static func snapshot(view: UIView, recording: Bool = false, description: String? = nil) {
+    let description = description ?? String(describing: type(of: view))
+    let frame = UIScreen.main.bounds
+    view.frame = frame
+    
+    let snapshot = view.snapshot()
+    guard let image = snapshot else { return }
+    let fileManager: FileManager = FileManager()
+    guard let dirPath = Bundle.main.infoDictionary?["UI_TEST_DIR"] as? String else { fatalError("UI_TEST_DIR not defined in your info.plist") }
+    var dirURL = URL(fileURLWithPath: dirPath)
+    if recording {
+      dirURL.appendPathComponent("/reference")
+    }
+    guard let pngData = UIImagePNGRepresentation(image) else { return }
+    let scaleFactor = Int(UIScreen.main.scale)
+    let fileURL = dirURL.appendingPathComponent("\(description)@\(scaleFactor)x.png")
+    guard let _ = try? fileManager.createDirectory(at: dirURL, withIntermediateDirectories: true, attributes: nil) else { return }
+    guard let _ = try? pngData.write(to: fileURL) else { return }
+  }
+  
+  
+}
+
+/// Test a ViewControllerModellableView embedded in a Container with a specific ViewModel.
+/// The test will produce a screenshot of the view.
+/// The screenshots will be located in the directory specified inside the plist with the `UI_TEST_DIR` key.
+/// After the screenshot is completed, the test will pass.
+/// This function can only be used in a XCTest environment.
+public func test<V: ViewControllerModellableView & UIView>(_ viewType: V.Type,
```

Comment:
I'm not totally sure about this name. 
`test` sounds really generic to me...  Can we use something like `testView` or `takeSnapshot`? I think that it describe better what is going on.

Reply:
testView becomes test(view, ...) following swift naming conventions. We could add ‘view’ as label of the first parameter, or we can leave it as is.